munit-hedgehog/shared/src/main/scala/munit/HedgehogSuite.scala

```diff
@@ -0,0 +1,115 @@
+package munit
+
+import hedgehog._
+import hedgehog.core.{PropertyConfig, PropertyT, Seed}
+import hedgehog.predef.Monad
+
+import munit.internal.Compat._
+
+/**
+ * Provides the ability to write property based tests using the Hedgehog library.
+ *
+ * Properties are defined by calling one of the various `property` or `propertyF`
+ * methods and passing a `PropertyT[Result]` or `PropertyT[F[Result]]` respectively.
+ * For example:
+ * {{{
+ * property("additive identity") {
+ *   Gen.int(Range.linearFrom(0, Int.MinValue, Int.MaxValue)).forAll.map { n =>
+ *     assertEquals(n + 0, n)
+ *   }
+ * }
+ * }}}
+ *
+ * Note both Hedgehog `Result` and munit assertions are supported.
+ */
+trait HedgehogSuite extends FunSuite {
+
+  /**
+   * Provides the default configuration for running property tests.
+   * This can be overriden to change the default or can be changed on
+   * a test-by-test basis by using an overload of `property` and `propertyF`.
+   */
+  lazy val propertyConfig: PropertyConfig = PropertyConfig.default
+  lazy val propertySeed: Long = System.currentTimeMillis()
+
+  def property(
+      name: String
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    property(name, propertyConfig)(prop)
+
+  def property(
+      options: TestOptions
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    property(options, propertyConfig)(prop)
+
+  def property(
+      name: String,
+      config: PropertyConfig
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    property(new TestOptions(name, Set.empty, loc), config)(prop)
+
+  def property(
+      options: TestOptions,
+      config: PropertyConfig
+  )(prop: PropertyT[Result])(implicit loc: Location): Unit =
+    test(options)(check(prop, config, propertySeed))
+
+  def propertyF[F[_]: Monad](
+      name: String
+  )(prop: PropertyT[F[Result]])(implicit loc: Location): Unit =
+    propertyF(name, propertyConfig)(prop)
+
+  def propertyF[F[_]: Monad](
+      options: TestOptions
+  )(prop: PropertyT[F[Result]])(implicit loc: Location): Unit =
+    propertyF(options, propertyConfig)(prop)
+
+  def propertyF[F[_]: Monad](
+      name: String,
+      config: PropertyConfig
+  )(prop: PropertyT[F[Result]])(implicit loc: Location): Unit =
+    propertyF(new TestOptions(name, Set.empty, loc), config)(prop)
+
+  def propertyF[F[_]: Monad](
+      options: TestOptions,
+      config: PropertyConfig
+  )(prop: PropertyT[F[Result]])(implicit loc: Location): Unit =
+    test(options)(checkF(prop, config, propertySeed))
+
+  /**
+   * Checks the supplied `Property[Result]`, throwing a `HedgehogFailException`
+   * if the property was falsified.
+   */
+  def check(
```

Comment:
would it make sense to make the `check` methods private?

Reply:
I made it public b/c in scodec-bits, I had some cases where I wanted to call `check` multiple times in a single test. That's not really necessary though so I'm happy with this being private.